### PR TITLE
Feature accepted requests

### PR DESCRIPTION
**Pull request by dtuckersuh**
Addresses Task #49 of Agile Board

- Add 'show accepted only' functionality to Request page that will filter all requests (sent and received) based on whether they were accepted or not

- Port in Rating Modal popup created by @CalebNale to add rating functionality after request is completed

- Add functionality to change 'completed' status of requests and update page to filter out completed requests

### CODE REVIEW

File: tutor4points/core/templates/home.html [hunk 1]
```diff
@@ -44,14 +44,21 @@ <h2>Tutor Requests</h2>
...
-                                            Accept
-                                        </button>
-                                        <button type="submit" name="decline-request" class="btn-secondary respond-button" 
-                                        data-toggle= "modal" data-target="#modal-decline-request">
-                                            Decline
-                                        </button>
+                                        {% if tutor_request.accepted %}
+                                            <button type="submit" name="btn-accept-request" cless="btn-secondary"
+                                                data-toggle="modal" data-target="#modal-complete-request">
+                                                Complete
```
Comment (thivnguyen) [inline tutor4points/core/templates/home.html:50]: I think a better button name would be "Completed and Paid" or something along that line, so the tutor knows that they should only click the button after they have fully complete and paid for their session. Another option is to just add a checkbox to the form as "Paid" and "Completed" and make sure the user check marks those 2 fields before submitting the form.

File: tutor4points/core/templates/home.html [hunk 2]
```diff
@@ -44,14 +44,21 @@ <h2>Tutor Requests</h2>
...
-                                            data-toggle= "modal" data-target="#modal-accept-request">
-                                            Accept
-                                        </button>
-                                        <button type="submit" name="decline-request" class="btn-secondary respond-button" 
-                                        data-toggle= "modal" data-target="#modal-decline-request">
-                                            Decline
-                                        </button>
+                                        {% if tutor_request.accepted %}
+                                            <button type="submit" name="btn-accept-request" cless="btn-secondary"
+                                                data-toggle="modal" data-target="#modal-complete-request">
```
Comment (thivnguyen) [inline tutor4points/core/templates/home.html:49]: It'll be helpful to still have the status next to the button or above it, so the user knows the request has been accepted.

**Review by thivnguyen — APPROVED** [review]
- The "Sent" box works great. It's good that the completed requests disappear right after. 
- The hidden fields send in the correct request-id and tutor-id too (I wasn't able to test for the "Inbox" portion because they don't appear)

Some Bugs:
- The accepted requests of Inbox don't show up like the "Sent" box
- When clicking the profiles, the links don't lead to the tutor profiles anymore.
- It'll be good to make the "tutee/tutor" field be a hidden input, but @CalebNale can work with that for the next sprint.

Overall, there are no major bugs, so it could be merged for this sprint. Or, it could be saved for the next sprint. You can decide whether you would like to merge it

File: tutor4points/core/templates/home.html [hunk 3]
```diff
@@ -44,14 +45,21 @@ <h2>Tutor Requests</h2>
...
                                     <div class = "respond-btns d-flex">
-                                        <button type="submit" name="accept-request" class="btn-secondary respond-button" 
-                                            data-toggle= "modal" data-target="#modal-accept-request">
-                                            Accept
-                                        </button>
-                                        <button type="submit" name="decline-request" class="btn-secondary respond-button" 
-                                        data-toggle= "modal" data-target="#modal-decline-request">
-                                            Decline
-                                        </button>
+                                        {% if tutor_request.accepted %}
```
Comment (thivnguyen) [inline tutor4points/core/templates/home.html:48]: I don't see the "Complete" button appearing on the homepage for accepted requests on my end.

File: tutor4points/core/templates/requests.html [hunk 4]
```diff
@@ -14,14 +14,26 @@
...
 {% block content %}
 
+<div class="show-accepted">
+    <form method="post">
+    {% csrf_token %}
+        <div class="form-check">
+            <input class="form-check-input" type="checkbox" value="" id="show-accepted" name="show-accepted"
+                onChange="this.form.submit()" {{accept_filter}}>
+            <label class="form-check-label" for="show-accepted">
+                Show Accepted
```
Comment (thivnguyen) [inline tutor4points/core/templates/requests.html:24]: I think it'll be helpful to have 2 checkboxes as discussed in our last team meeting.  One for the "inbox" and one for the "sent".

**Review by thivnguyen — APPROVED** [review]
Everything looks good on the requests page! All the ids seem to be sent correctly, and forms are submitted. 

It'll be great if you could do 2 separate checkboxes as discussed in our last team meeting, one for "Inbox" and one for "Sent" so the user can filter separately.

The only bug I found was that the "complete" button does not appear for me on the homepage for any of the requests, so I was not able to test the functionality of that.

Overall, there is 1 minor bug that could be fixed quickly and one small change suggested. No major bugs so code could be merged.

File: tutor4points/core/views.py [hunk 5]
```diff
@@ -289,9 +290,52 @@ def requests(request, id):
...
+                count = 0
+                sum = 0
+                for i in tutor_ratings:
+                    count += 1
+                    sum += i.rating
+                userGivenTo.tutee_avg_rating = sum/count
+            userGivenTo.save()
+
+            currentRequest = TutorRequest.objects.get(pk=request.POST['request-id'])
+            currentRequest.completed = True
```
Comment (cathyvu1) [inline tutor4points/core/views.py:329]: When a tutor marks a request as complete and submits a rating, the request will also disappear on the tutee's side before the tutee has submitted a rating (vice versa). I think we need to rethink how to check when a request is completed by both sides. One solution is to have a tutor_rating_completed and tutee_rating_completed for each request and then we can change completed to true when both tutor_rating_completed and tutee_rating_completed are true. Not sure if this is the best method though.

File: tutor4points/core/templates/requests.html [hunk 6]
```diff
@@ -14,14 +14,26 @@
 
 {% block content %}
 
+<div class="show-accepted">
+    <form method="post">
+    {% csrf_token %}
+        <div class="form-check">
+            <input class="form-check-input" type="checkbox" value="" id="show-accepted" name="show-accepted"
```
Comment (cathyvu1) [inline tutor4points/core/templates/requests.html:21]: Not a bug, but it would be nice to add styling to the checkbox so it matches the theme.

**Review by cathyvu1 — APPROVED** [review]
Overall functionality for accepted requests looks good.

**Pull request merged**